ssl/statem/statem_lib.c

```diff
@@ -1679,11 +1679,22 @@ int ssl_check_version_downgrade(SSL *s)
  */
 int ssl_set_version_bound(int method_version, int version, int *bound)
 {
+    int valid_tls;
+    int valid_dtls;
+
     if (version == 0) {
         *bound = version;
         return 1;
     }
 
+    valid_tls = version >= SSL3_VERSION && version <= TLS_MAX_VERSION_INTERNAL;
+    valid_dtls =
+        DTLS_VERSION_LE(version, DTLS_MAX_VERSION_INTERNAL) &&
+        DTLS_VERSION_GE(version, DTLS1_BAD_VER);
```

Comment:
A naïve question: why allow equality vs `DTLS1_BAD_VER`?

Reply:
> A naïve question: why allow equality vs `DTLS1_BAD_VER`?

Because it isn't actually a "bad version", rather it is a non-standard version used in some IIRC Cisco products, and still supported by OpenSSL, though we don't have a string limit that matches this version, at the API level, users can specify the constant. @mattcaswell may be able to confirm/deny.

Reply:
Okay, makes sense.

Reply:
I can "confirm" @vdukhovni's description of `DTLS1_BAD_VER` above